modules/messages/tests/server/message-to-influx.server.service.tests.js

```diff
@@ -0,0 +1,201 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+var mongoose = require('mongoose'),
+    path = require('path'),
+    async = require('async'),
+    User = mongoose.model('User'),
+    Message = mongoose.model('Message'),
+    messageToInfluxService =
+  require(path.resolve('./modules/messages/server/services/message-to-influx.server.service'));
+require('should');
+
+/**
+ * Unit tests
+ */
+describe('Message to influx server service Unit Tests:', function() {
+  var user1,
+      user2;
+
+  // here we create the users
```

Comment:
@simison I get that theory, and in the context of jenkins it's probably true in practice. But nonetheless, this is a very fragile approach to testing. There are many different things under test here, so this is a "functional test" and not a "unit test". These tests will fail under all kinds of scenarios through no fault of the actual code.

I'm not deep enough into the whole tr codebase to know what the testing strategy is, but in general, I'd recommend mocking and stubbing over this type of test. Tools like `sinon` make it fairly easy to produce mocks. Plus, they can often be shared application wide. Although it can require a different architecture to get them into your code under test.

On a totally separate note, I'm not sure how much testing we really need on the InfluxDB stuff. It's a non core service, so if it fails, unless it affects the running of the site, it's an non critical issue.